### PR TITLE
fix: Improve oAuth option handling

### PR DESCRIPTION
**Pull request by daffl**
Pull request that should close https://github.com/feathersjs/feathers/issues/1324 specifically

- `9)` by adding an `expressSession` option to provide a custom session store
- `10)` by using both `profile.sub` and `profile.id`
- `11)` by adding `getRedirect` to the strategy (so it is also customizable on a per-strategy basis)
- `12)` by supporting the `getEntityQuery` and `getEntityData` methods that should make customization easier

@KidkArolis let me know if you see anything that's missing

### CODE REVIEW

File: packages/authentication-oauth/src/utils.ts [hunk 1]
```diff
@@ -1,38 +1,24 @@
...
 
 export interface OauthSetupSettings {
-  path?: string;
-  authService?: string;
-  linkStrategy?: string;
-  getRedirect? (service: AuthenticationService, data: AuthenticationResult|Error): Promise<string>;
+  path: string;
+  authService: string;
+  linkStrategy: string;
+  expressSession: RequestHandler;
```
Comment (KidkArolis) [inline packages/authentication-oauth/src/utils.ts]: That doesn't look like the right type?

Comment (daffl) [reply packages/authentication-oauth/src/utils.ts]: It's `import { RequestHandler } from 'express';` right? It at least doesn't complain when initializing the default options in https://github.com/feathersjs/feathers/pull/1335/files/40ebbddeae3995e840c059146a6e3d3597087107#diff-b3c31ade33c8365501f61c7361f117b2R17

File: packages/authentication-oauth/src/strategy.ts [hunk 2]
```diff
@@ -29,6 +30,18 @@ export class OAuthStrategy extends AuthenticationBaseStrategy {
...
 
+  async getEntityQuery (profile: OAuthProfile, _params: Params) {
+    return {
+      [`${this.name}Id`]: profile.sub || profile.id
+    };
+  }
+
+  async getEntityData (profile: OAuthProfile, _existingEntity: any, _params: Params) {
+    return {
+      [`${this.name}Id`]: profile.id
```
Comment (KidkArolis) [inline packages/authentication-oauth/src/strategy.ts]: Is profile.sub missing here?

File: packages/authentication-oauth/src/utils.ts (hunk 1 above)
Comment (KidkArolis) [reply packages/authentication-oauth/src/utils.ts]: Ah, I guess this makes sense if expressSession is like a middleware (RequestHandler).

File: packages/authentication-oauth/src/strategy.ts (hunk 2 above)
Comment (daffl) [reply packages/authentication-oauth/src/strategy.ts]: Good catch. Fixed!

**Pull request merged**